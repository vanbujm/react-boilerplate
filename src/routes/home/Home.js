/**
 * React Starter Kit (https://www.reactstarterkit.com/)
 *
 * Copyright Â© 2014-present Kriasoft, LLC. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

import React from 'react';
import PropTypes from 'prop-types';
import withStyles from 'isomorphic-style-loader/lib/withStyles';
import { connect } from 'react-redux';
import { retrieveReactNews } from '../../actions/news';
import s from './Home.css';

export class Home extends React.Component {
  static propTypes = {
    retrieveReactNews: PropTypes.func.isRequired,
    news: PropTypes.arrayOf(
      PropTypes.shape({
        title: PropTypes.string.isRequired,
        link: PropTypes.string.isRequired,
        content: PropTypes.string,
      }),
    ),
  };

  componentDidMount() {
    this.props.retrieveReactNews();
  }

  render() {
    const renderedNews = Array.isArray(this.props.news)
      ? this.props.news.map(item => (
          <article key={item.link} className={s.newsItem}>
            <h1 className={s.newsTitle}>
              <a href={item.link}>{item.title}</a>
            </h1>
            <div
              className={s.newsDesc}
              // eslint-disable-next-line react/no-danger
              dangerouslySetInnerHTML={{ __html: item.content }}
            />
          </article>
        ))
      : null;

    return (
      <div className={s.root}>
        <div className={s.container}>
          <h1>React.js News</h1>
          {renderedNews}
        </div>
      </div>
    );
  }
}

Home.defaultProps = {
  news: [],
};

const mapStateToProps = store => ({
  news: store.news.reactNews,
});
const mapDispatchToProps = {
  retrieveReactNews,
};

export default connect(mapStateToProps, mapDispatchToProps)(
  withStyles(s)(Home),
);
