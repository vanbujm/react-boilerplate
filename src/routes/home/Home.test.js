/* eslint-env jest */
/* eslint-disable padded-blocks, no-unused-expressions */

import React from 'react';
import renderer from 'react-test-renderer';
import { Home } from './Home';

const mockNewsData = [
  {
    content: '<h1>Some mock content with html </h1>',
    link: 'https://reactjsnews.com/fake-news',
    title: 'Using Proxies with Redux Types',
  },
];

const newsActionSpy = jest.fn();

const mockNewsAction = () => newsActionSpy();

describe('Home Page', () => {
  test('renders', () => {
    const renderedComponent = renderer
      .create(<Home requestReactNews={mockNewsAction} />)
      .toJSON();

    expect(newsActionSpy.mock.calls.length).toBeGreaterThan(0);
    expect(renderedComponent).toMatchSnapshot();
  });

  test('renders news content', () => {
    const renderedComponent = renderer
      .create(<Home requestReactNews={mockNewsAction} news={mockNewsData} />)
      .toJSON();

    expect(renderedComponent).toMatchSnapshot();
  });
});
