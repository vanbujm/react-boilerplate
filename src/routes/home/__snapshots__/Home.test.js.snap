// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page renders 1`] = `
<div
  className="root"
>
  <div
    className="container"
  >
    <h1>
      React.js News
    </h1>
  </div>
</div>
`;

exports[`Home Page renders news content 1`] = `
<div
  className="root"
>
  <div
    className="container"
  >
    <h1>
      React.js News
    </h1>
    <article
      className="newsItem"
    >
      <h1
        className="newsTitle"
      >
        <a
          href="https://reactjsnews.com/fake-news"
        >
          Using Proxies with Redux Types
        </a>
      </h1>
      <div
        className="newsDesc"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<h1>Some mock content with html </h1>",
          }
        }
      />
    </article>
  </div>
</div>
`;
